airflow/contrib/hooks/gcp_vision_hook.py

```diff
@@ -0,0 +1,245 @@
+                                        product_set=product_set, update_mask=update_mask, retry=retry,
+                                        timeout=timeout, metadata=metadata)
+        self.log.info('ProductSet updated: %s', response.name if response else '')
+        self.log.debug('ProductSet updated:\n%s', response)
+        return MessageToJson(response)
+
+    @GoogleCloudBaseHook.fallback_to_default_project_id
+    def delete_product_set(self, location, product_set_id, project_id=None, retry=None, timeout=None,
+                           metadata=None):
+        """
+        For the documentation see:
+        :py:class:`~airflow.contrib.operators.gcp_vision_operator.CloudVisionProductSetDeleteOperator`
+        """
+        client = self.get_client()
+        name = PRODUCT_SET_NAME_TEMPLATE.format(project_id, location, product_set_id)
+        self.log.info('Deleting ProductSet: %s', name)
+        response = self._handle_request(lambda **kwargs: client.delete_product_set(**kwargs), name=name,
+                                        retry=retry, timeout=timeout, metadata=metadata)
+        self.log.info('ProductSet with the name [%s] deleted.', name)
+        return response
+
+    @staticmethod
+    def _determine_name(entity, entity_id, location, project_id):
+        is_product = type(entity) == Product
+        entity_label = 'Product' if is_product else 'ProductSet'
+        entity_id_label = 'product_id' if is_product else 'productset_id'
+        entity_name_template = PRODUCT_NAME_TEMPLATE if is_product else PRODUCT_SET_NAME_TEMPLATE
+        explicit_name = getattr(entity, 'name')
+        if not location or not entity_id:
+            # Not enough parameters to construct the name. Trying to use the name from Product.
+            if not explicit_name:
+                raise AirflowException("Unable to determine the {} name. Please either set the name "
+                                       "directly in the {} object or provide the `location` and "
+                                       "`{}` parameters.".format(entity_label, entity_label, entity_id_label))
+        else:
+            template_name = entity_name_template.format(project_id, location, entity_id)
+            if not explicit_name:
+                entity.name = template_name
+            elif explicit_name != template_name:
+                raise AirflowException("The {} name provided in the object ({}) is different than the"
+                                       " name created from the input parameters ({}). Please either: "
+                                       "1) Remove the {} name, 2) Remove the location and "
+                                       "{} parameters, 3) Unify the {} name and input "
+                                       "parameters.".format(entity_label, explicit_name, template_name,
+                                                            entity_label, entity_id_label, entity_label))
+        return entity
```

Comment:
It's required?

Reply:
Yes, because when invoking the function:
`product = self._determine_name(product, product_id, location, project_id)`